### PR TITLE
chore: add default template generation

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Sorry for big PR. This is incorporating the following changes:

* arguments support
* stop supporting version as env, add it as argument
* generating default configuration
  * support for install token as env and argument
* fail if link to binary is incorrect

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
I recommend to read last part of the script without diff (template generation)

**Review by astencel-sumo — APPROVED** [review]
I believe you!

**Comment by sumo-drosiek** [comment]
@astencel-sumo
> I believe you!

I don't believe in this code, so you neither :sweat:

File: scripts/install.sh [hunk 1]
```diff
@@ -2,8 +2,161 @@
...
+  cat << EOF
+
+Usage: bash install.sh [--${ARG_LONG_TOKEN} <token>] [--${ARG_LONG_COLLECTOR} name] [--${ARG_LONG_TAG} <key>=<value> [ --${ARG_LONG_TAG} ...]] [--${ARG_LONG_API} <url>] [--${ARG_LONG_CONFIG} <config dir path>] [--${ARG_LONG_STORAGE} <storage dir path>] [--${ARG_LONG_VERSION} <version>] [--${ARG_LONG_YES}] [--${ARG_LONG_VERSION} <version>] [--${ARG_LONG_HELP}]
+  -${ARG_SHORT_TOKEN}, --${ARG_LONG_TOKEN} <token>     Installation token
+  -${ARG_SHORT_COLLECTOR}, --${ARG_LONG_COLLECTOR} <name>          Collector name (default is your hostname)
+  -${ARG_SHORT_TAG}, --${ARG_LONG_TAG} <key=value>                Tag in format key=value
+
+  -${ARG_SHORT_API}, --${ARG_LONG_API} <url>                      Api URL
+  -${ARG_SHORT_CONFIG}, --${ARG_LONG_CONFIG} <config dir path>       Path to the storage directory (default is '/var/lib/sumologic/file_storage')
+  -${ARG_SHORT_STORAGE}, --${ARG_LONG_STORAGE} <storage dir path>     Path to the configuration directory (default is '/etc/otelcol-sumo')
```
Comment (perk-sumo) [inline scripts/install.sh:55]: Option and description seems to be switched for those two

**Review by perk-sumo — APPROVED** [review]
LGTM except for one problem in the usage() function

**Pull request merged**